libc-top-half/musl/src/stat/futimesat.c

```diff
@@ -11,7 +13,12 @@ int __futimesat(int dirfd, const char *pathname, const struct timeval times[2])
 		int i;
 		for (i=0; i<2; i++) {
 			if (times[i].tv_usec >= 1000000ULL)
+#ifdef __wasilibc_unmodified_upstream // WASI has no syscall
 				return __syscall_ret(-EINVAL);
```

Comment:
I wonder if this a common pattern if it would be worth at least defining `__syscall_ret` macro?

Reply:
I'm not necessarily opposed to it, but it feels a little awkward, because either:
 - It introduces the Linux-specific convention of using negative values in [-4095,0) to represent errors, or
 - It doesn't use this, which risks having it behave differently than the code calling it expects.